bindings/node/lib/clientEncryption.js

```diff
@@ -28,6 +28,40 @@ module.exports = function (modules) {
    * @property {string} [gcp.endpoint] If present, a host with optional port. E.g. "example.com" or "example.com:443". Defaults to "oauth2.googleapis.com"
    */
 
+  /**
+   * @typedef {object} DataKey A data key as stored in the database.
+   * @property {UUID} _id A unique identifier for the key.
+   * @property {number} version A numeric identifier for the schema version of this document. Implicitly 0 if unset.
+   * @property {string[]} [keyAltNames] Alternate names to search for keys by. Used for a per-document key scenario in support of GDPR scenarios.
+   * @property {Binary} keyMaterial Encrypted data key material, BinData type General.
+   * @property {Date} creationDate The datetime the wrapped data key material was imported into the Key Database.
+   * @property {Date} updateDate The datetime the wrapped data key material was last modified. On initial import, this value will be set to creationDate.
+   * @property {number} status 0 = enabled, 1 = disabled
+   * @property {object} masterKey the encrypted master key
+   */
+
+  /**
+   * @typedef {string} KmsProvider A string containing the name of a kms provider.  Valid options are 'aws', 'azure', 'gcp', 'kmip', or 'local'
+   */
+
+  /**
+   * @typedef { object } ClientSession The ClientSession class from the MongoDB Node driver (see https://mongodb.github.io/node-mongodb-native/4.8/classes/ClientSession.html)
+   */
+
+  /**
+   * @typedef {object} DeleteResult The result of a delete operation from the MongoDB Node driver (see https://mongodb.github.io/node-mongodb-native/4.8/interfaces/DeleteResult.html)
+   * @property {boolean} acknowledged Indicates whether this write result was acknowledged. If not, then all other members of this result will be undefined.
+   * @property {number} deletedCount The number of documents that were deleted
+   */
+
+  /**
+   * @typedef {object} BulkWriteResult The BulkWriteResult class from the MongoDB Node driver (https://mongodb.github.io/node-mongodb-native/4.8/classes/BulkWriteResult.html)
+   */
+
+  /**
+   * @typedef {object} FindCursor The FindCursor class from the MongoDB Node driver (see https://mongodb.github.io/node-mongodb-native/4.8/classes/FindCursor.html)
```

Comment:
jsdoc doesn't let us import types from the index.d.ts for documentation purposes (i.e., `import('../index.d.ts').FindCursor`), so I created dummy types for the purposes documentation.  mongodb and mognodb-client-encryption are peer dependencies, so no users will be using this without also having mongodb installed, I think linking to the mongodb docs is sufficient until we can convert this to TS and use TSDoc or figure out a better documentation solution.